lib/plugins/features/movies/controller.js

```diff
@@ -36,5 +40,42 @@ exports.retrieve = async (queryParams) => {
     } else {
       qb.orderBy('release_year');
     }
-  }).fetchAll();
+  }).fetchAll(MOVIE_MODEL_FETCH_OPTIONS);
+};
+
+function toTitleCase (str) {
+  return str.replace(/\w\S*/g, (txt) => {
+    return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase();
+  });
+}
+
+exports.addLocationsToMovies = async (payload, movieId) => {
+  return new Movie({ id: movieId })
+  .fetch(MOVIE_MODEL_FETCH_OPTIONS)
+  .then((movie) => {
+    if (movie === null) {
```

Comment:
You can fetch with the `{ required: true }` option so that an error is thrown if the return value is null. That way you could simply catch the error and throw the custom `NotFoundError`!